subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/configurations/renderer/ConsoleConfigurationReportRenderer.java

```diff
@@ -0,0 +1,452 @@
+        if (hasAnyRelevantConfigs) {
+            if (spec.isSearchForSpecificVariant()) {
+                writeSearchResults(spec, data);
+            } else {
+                if (spec.isShowLegacy()) {
+                    writeLegacyResults(spec, data);
+                } else {
+                    writeNonLegacyResults(spec, data);
+                }
+            }
+        } else {
+            writeCompleteAbsenceOfResults(spec, data);
+        }
+    }
+
+    private void writeCompleteAbsenceOfResults(AbstractConfigurationReportSpec spec, ConfigurationReportModel data) {
+        message("There are no " + spec.getFullReportedTypeDesc() + "s (including legacy " + spec.getReportedTypeAlias() + "s) present in project '" + data.getProjectName() + "'.");
+    }
+
+    private void writeSearchResults(AbstractConfigurationReportSpec spec, ConfigurationReportModel data) {
+        Optional<ReportConfiguration> searchResult = data.getConfigNamed(spec.getSearchTarget().get());
+        if (searchResult.isPresent()) {
+            writeResults(spec, data, () -> Collections.singletonList(searchResult.get()));
+        } else {
+            message("There are no " + spec.getFullReportedTypeDesc() + "s on project '" + data.getProjectName() + "' named '" + spec.getSearchTarget().get() + "'.");
+        }
+    }
+
+    private void writeLegacyResults(AbstractConfigurationReportSpec spec, ConfigurationReportModel data) {
+        final Supplier<List<ReportConfiguration>> legacySupplier = () -> data.getAllConfigs().stream()
+            .filter(c -> c.isLegacy() || spec.isPurelyCorrectType(c))
+            .collect(Collectors.toList());
+        writeResults(spec, data, legacySupplier);
+    }
+
+    private void writeNonLegacyResults(AbstractConfigurationReportSpec spec, ConfigurationReportModel data) {
+        final List<ReportConfiguration> nonLegacyConfigs = data.getAllConfigs().stream()
+            .filter(spec::isPurelyCorrectType)
+            .collect(Collectors.toList());
+        if (nonLegacyConfigs.isEmpty()) {
+            message("There are no purely " + spec.getReportedConfigurationDirection() + " " + spec.getReportedTypeAlias() + "s present in project '" + data.getProjectName() + "'.");
+
+            final boolean hasLegacyConfigs = data.getAllConfigs().stream().anyMatch(ReportConfiguration::isLegacy);
+            if (hasLegacyConfigs) {
+                message("Re-run this report with the '--all' flag to include legacy " + spec.getReportedTypeAlias() + "s (legacy = consumable and resolvable).");
+            }
+        } else {
+            writeResults(spec, data, () -> nonLegacyConfigs);
+        }
+    }
+
+    private void writeResults(AbstractConfigurationReportSpec spec, ConfigurationReportModel data, Supplier<List<ReportConfiguration>> configsSupplier) {
+        final List<ReportConfiguration> configs = configsSupplier.get();
```

Comment:
I don't understand the point of using a `Supplier` here only to resolve it immediately.

Reply:
Looks like there is no point currently, another artifact of refactoring to prune.